corehq/apps/case_importer/suggested_fields.py

```diff
@@ -0,0 +1,88 @@
+from django.utils.translation import ugettext as _
+import itertools
+from corehq.apps.case_importer.util import get_case_properties_for_case_type, \
+    RESERVED_FIELDS
+from dimagi.ext import jsonobject
+
+
+def _combine_field_specs(field_specs, exclude_fields):
+    """
+    take a list of FieldSpec objects and return a sorted list where field is unique
+    and fields in exclude_fields are removed, and where the first mention of a
+    field in field_specs will win out over any repeats.
```

Comment:
This behavior feels a little cryptic, couldn't you just use `dict.update` and apply `special_field_specs` last?